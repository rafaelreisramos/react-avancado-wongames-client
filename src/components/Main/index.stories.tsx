import { Story, Meta } from '@storybook/react/types-6-0'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'Next.js + Storybook boilerplate!'
  }
} as Meta

export const Default: Story = (args) => <Main {...args} />

export const Basic: Story = (args) => <Main {...args} />
Basic.args = {
  title: 'Next.js boilerplate with storybook!'
}
