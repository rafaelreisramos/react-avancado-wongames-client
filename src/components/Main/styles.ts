import styled from 'styled-components'

export const Container = styled.main`
  height: 100vh;
  width: 100vw;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #c0c0c0;
`

export const LogoContainer = styled.div`
  display: flex;
  width: 350px;
  align-items: center;
  justify-content: space-between;
`

export const Title = styled.h1`
  font-size: 3.2rem;
  color: ${(props) => props.theme.colors.white};
`
