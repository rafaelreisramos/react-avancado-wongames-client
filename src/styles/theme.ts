export default {
  colors: {
    mainBg: '#06092b',
    white: '#fafafa'
  }
}
