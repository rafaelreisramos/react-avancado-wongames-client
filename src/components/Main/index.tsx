import * as S from './styles'

const Main = ({ title = 'Next.js boilerplate!' }) => (
  <S.Container>
    <S.LogoContainer>
      <img
        src="./img/nextjs-black-logo.svg"
        alt="Logo Next"
        width={150}
        height={150}
      />
      <img
        src="./img/storybook-default-logo.svg"
        alt="Storybook Next"
        width={150}
        height={150}
      />
    </S.LogoContainer>
    <S.Title>{title}</S.Title>
  </S.Container>
)

export default Main
