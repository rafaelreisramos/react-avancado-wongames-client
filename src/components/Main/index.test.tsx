import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import theme from '../../styles/theme'
import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Main />
      </ThemeProvider>
    )

    expect(
      screen.getByRole('heading', { name: /next.js boilerplate/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /next.js boilerplate/i })
    ).toHaveStyle(`color: ${theme.colors.white}`)

    expect(container.firstChild).toMatchSnapshot()
  })
})
