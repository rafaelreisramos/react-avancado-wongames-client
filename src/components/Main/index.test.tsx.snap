// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  height: 100vh;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #c0c0c0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 350px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  font-size: 3.2rem;
  color: #fafafa;
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Logo Next"
      height="150"
      src="./img/nextjs-black-logo.svg"
      width="150"
    />
    <img
      alt="Storybook Next"
      height="150"
      src="./img/storybook-default-logo.svg"
      width="150"
    />
  </div>
  <h1
    class="c2"
  >
    Next.js boilerplate!
  </h1>
</main>
`
